GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -425,13 +425,19 @@ public void LoadingCompleted()
                 _loadedToBeSelectedRevisionsCount = ToBeSelectedObjectIds.Count;
             }
 
-            if (_loadedToBeSelectedRevisionsCount > 0 && _revisionGraph.Count > 0)
+            if (_revisionGraph.Count == 0)
+            {
+                MarkAsDataLoadingComplete();
+            }
+            else
             {
                 // Rows have not been selected yet
                 ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
                 {
                     await this.SwitchToMainThreadAsync();
 
+                    SetRowCountAndSelectRowsIfReady();
```

Comment:
This line vs old 466 is the actual change in the PR.
Is it OK to update RowCount before rendering is initiated?

Reply:
> Is it OK to update RowCount before rendering is initiated?

Yes, it is. `RowCount` is unconditionally set to the number of loaded revisions.
When rendering is requested, `CacheTo()` is called. This function creates and sorts the graph segments as far a needed.